opendevin/config.py

```diff
@@ -37,44 +37,18 @@
         config[k] = tomlConfig[k]
 
 
-def _get(key: str, default):
-    value = config.get(key, default)
-    if not value:
-        value = os.environ.get(key, default)
-    return value
-
-
-def get_or_error(key: str):
+def get(key: str, default=None, required=False):
```

Comment:
It seems there might have been a misunderstanding. The change from get_or_error(key: str) to get(key: str, default=None, required=False) does change the behavior slightly, as the latter allows for a default value to be specified and does not raise an error when the key is not found if required is set to False. However, if required is set to True in get(key, required=True), the behavior is equivalent to get_or_error(key) because it will raise an error when the key is not found.

Reply:
I agree with @enyst--we shouldn't have a public `get` that allows passing in a default value. Defaults should always be taken from `config.py`.

What we want to avoid is
* file1.py: `get('FOO', 1)`
* file2.py: `get('FOO', 2)`

If that happens, there are two different values of `FOO` being used throughout the app. We want to _always_ enforce the same default.

Reply:
@rbren fixed